core/src/stored/backends/crud_storage.h

```diff
@@ -0,0 +1,61 @@
+/*
+   BAREOS® - Backup Archiving REcovery Open Sourced
+
+   Copyright (C) 2024-2024 Bareos GmbH & Co. KG
+
+   This program is Free Software; you can redistribute it and/or
+   modify it under the terms of version three of the GNU Affero General Public
+   License as published by the Free Software Foundation and included
+   in the file LICENSE.
+
+   This program is distributed in the hope that it will be useful, but
+   WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+   Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301, USA.
+*/
+
+#ifndef BAREOS_STORED_BACKENDS_CRUD_STORAGE_H_
+#define BAREOS_STORED_BACKENDS_CRUD_STORAGE_H_
+
+#include <map>
+#include <string_view>
+#include <unordered_map>
+#include <gsl/span>
+#include <cstdint>
+#include "lib/bstringlist.h"
+#include "tl/expected.hpp"
+
+class CrudStorage {
+  struct Stat {
+    size_t size{0};
+  };
+  std::string m_program{"/bin/false"};
+  uint32_t m_program_timeout{30};
```

Comment:
This should probably be `std::chrono::seconds` to make clear that this is in seconds.